pgmpy/factors/Factor.py

```diff
@@ -466,12 +466,14 @@ def gen():
                     self.values[value_index])
                 string_list.append(html_string)
             else:
-                string = "%s\t\t%4.4f" % ("\t\t".join(prob_list),
-                                          self.values[value_index])
-                string_list.append(string)
+                prob_list.append(self.values[value_index])
+                string_list.append(prob_list)
             value_index += 1
 
-        return "\n".join(string_list)
+        if html:
+            return "\n".join(string_list)
+        else:
+            return tabulate(string_list, tablefmt="grid")
```

Comment:
Please replace the ```html``` part of the code using ```tabulate```` as well. As the ```html``` representation is pretty simple in case of ```Factor```, I think we can use ```tabulate``` here.